Lib/traceback.py

```diff
@@ -490,14 +491,19 @@ def __init__(self, exc_type, exc_value, exc_traceback, *, limit=None,
             cause = None
         if (exc_value and exc_value.__context__ is not None
             and id(exc_value.__context__) not in _seen):
-            context = TracebackException(
-                type(exc_value.__context__),
-                exc_value.__context__,
-                exc_value.__context__.__traceback__,
-                limit=limit,
-                lookup_lines=False,
-                capture_locals=capture_locals,
-                _seen=_seen)
+            try:
+                context = TracebackException(
+                    type(exc_value.__context__),
+                    exc_value.__context__,
+                    exc_value.__context__.__traceback__,
+                    limit=limit,
+                    lookup_lines=False,
+                    capture_locals=capture_locals,
+                    _seen=_seen)
+            except RecursionError:
+                warnings.warn(
```

Comment:
I wouldn't emit a warning here. Save the fact the this happened in a Boolean field here and emit the warning in format().  Make it part of the formatted output rather than using the warnings module for it (which might be writing to a different place).

Reply:
done